Silicon/ApollolakePkg/Library/FirmwareUpdateLib/FirmwareUpdateLib.c

```diff
@@ -36,92 +36,7 @@
 
 #define FLASH_MAP_IN_FV_OFFSET   0xA4
 
-SPI_FLASH_SERVICE   *mFwuSpiService = NULL;
-
-/**
-  This function initialized boot media.
-
-  It initializes SPI services and SPI Flash size information.
-
-**/
-VOID
-EFIAPI
-InitializeBootMedia (
-  VOID
-  )
-{
-  mFwuSpiService = (SPI_FLASH_SERVICE *)GetServiceBySignature (SPI_FLASH_SERVICE_SIGNATURE);
-  if (mFwuSpiService == NULL) {
-    return;
-  }
-
-  mFwuSpiService->SpiInit ();
-}
-
-/**
-  This function reads blocks from the SPI device.
-
-  @param[in]  Address             The block address in the FlashRegionAll to read from on the SPI.
-  @param[in]  ByteCount           Size of the Buffer in bytes.
-  @param[out] Buffer              Pointer to caller-allocated buffer containing the data received during the SPI cycle.
-
-  @retval EFI_SUCCESS             Read completes successfully.
-  @retval others                  Device error, the command aborts abnormally.
-
-**/
-EFI_STATUS
-EFIAPI
-BootMediaRead (
-  IN     UINT64                  Address,
-  IN     UINT32                  ByteCount,
-  OUT    UINT8                   *Buffer
-  )
-{
-  return mFwuSpiService->SpiRead (FlashRegionBios, (UINT32)Address, ByteCount, Buffer);
-}
-
-/**
-  This function writes blocks from the SPI device.
-
-  @param[in]   Address            The block address in the FlashRegionAll to read from on the SPI.
-  @param[in]   ByteCount          Size of the Buffer in bytes.
-  @param[out]  Buffer             Pointer to the data to write.
-
-  @retval EFI_SUCCESS             Write completes successfully.
-  @retval others                  Device error, the command aborts abnormally.
-
-**/
-EFI_STATUS
-EFIAPI
-BootMediaWrite (
-  IN     UINT64                  Address,
-  IN     UINT32                  ByteCount,
-  OUT    UINT8                   *Buffer
-  )
-{
-  return mFwuSpiService->SpiWrite (FlashRegionBios, (UINT32)Address, ByteCount, Buffer);
-}
-
-/**
-  This function erases blocks from the SPI device.
-
-  @param[in]  Address             The block address in the FlashRegionAll to read from on the SPI.
-  @param[in]  ByteCount           Size of the region to erase in bytes.
-
-  @retval EFI_SUCCESS             Erase completes successfully.
-  @retval others                  Device error, the command aborts abnormally.
-
-**/
-EFI_STATUS
-EFIAPI
-BootMediaErase (
-  IN     UINT64                  Address,
-  IN     UINT32                  ByteCount
-  )
-{
-  return mFwuSpiService->SpiErase (FlashRegionBios, (UINT32)Address, ByteCount);
-}
-
+extern SPI_FLASH_SERVICE   *mFwuSpiService;

```

Comment:
Looks there are multiple extern variables. Can we use an API to get SpiService instance instead of extern variable?
```
ex)
GetFwuSpiServiceInstance () {
  if (mFwuSpiService == NULL) {
    mFwuSpiService = (SPI_FLASH_SERVICE *)GetServiceBySignature (SPI_FLASH_SERVICE_SIGNATURE);	
    if (mFwuSpiService != NULL) {
      mFwuSpiService->SpiInit ();
    }
  }
  return mFwuSpiService;
}
```

Otherwise, we need to wrap all necessary APIs. Some code is using APIs, but some are using mFwuSpiService directly.

Reply:
Removed externs in the latest patch